File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length;
        int n=matrix[0].length;
        int e=0;
        int f=n-1;
        while(f>=0 && e<m){
            if(matrix[e][f]==target) return true;
            if(matrix[e][f]>target) f--;
            else e++;
        }
        return false;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        // Brute force approach
        int m=matrix.size();
        int n=matrix[0].size();
        for(int e=0;e<m;e++){
            for(int f=0;f<n;f++){
                if(matrix[e][f]==target) return true;
            }
        }
        return false;
    }
};